File: Cs/Solution/MaxFreqSum.cs
using System;

namespace Solution
{
	public partial class Solution
	{
		public int MaxFreqSum(string s)
		{
			int[] freq = new int[26];
			int max_vowel = 0;
			int max_cons = 0;
			foreach (char c in s)
			{
				int i = c - 'a';
				freq[i]++;
				if ((c == 'a') || (c == 'e') || (c == 'i') || (c == 'o') || (c == 'u'))
				{
					max_vowel = Math.Max(max_vowel, freq[i]);
				}
				else
				{
					max_cons = Math.Max(max_cons, freq[i]);
				}
			}
			return max_vowel + max_cons;
		}
	}
}

File: Ruby/solution/max_freq_sum.rb
# @param {String} s
# @return {Integer}
def max_freq_sum(s)
  freq = Hash.new(0)
  max_vowel = 0
  max_consonant = 0
  s.each_char do |c|
    freq[c] += 1
    if "aeiou".include?(c)
      max_vowel = [max_vowel, freq[c]].max
    else
      max_consonant = [max_consonant, freq[c]].max
    end
  end
  max_vowel + max_consonant
end
